src/ur/dashboard_client.cpp

```diff
@@ -178,65 +195,313 @@ bool DashboardClient::retryCommand(const std::string& requestCommand, const std:
   return false;
 }
 
+void DashboardClient::parseSWVersion(int result[4], const std::string& input)
+{
+  std::istringstream parser(input);
+  parser >> result[0];
+  for (int idx = 1; idx < 4; idx++)
+  {
+    parser.get();  // Skip period
+    parser >> result[idx];
+  }
+}
+
+bool DashboardClient::lessThanVersion(const std::string& a, const std::string& b, const std::string& c)
+{
+  std::string ref = e_series_ == true ? a : b;
+  int parsedRef[4], parsedC[4];
+  parseSWVersion(parsedRef, ref);
+  parseSWVersion(parsedC, c);
+  bool ret = std::lexicographical_compare(parsedRef, parsedRef + 4, parsedC, parsedC + 4);
+  if (!ret)
+  {
+    throw UrException("Polyscope software version required is: " + ref + ", but actual version is: " + c);
```

Comment:
Will suggest that we throw the exception in the "command" function to make it obvious witch one is the one that is not supported by the robot or pass the command name to this function